src/compiler/builder.ts

```diff
@@ -312,29 +289,35 @@ namespace ts {
         state.program = undefined;
     }
 
-    /**
-     * Creates a clone of the state
-     */
-    function cloneBuilderProgramState(state: Readonly<BuilderProgramState>): BuilderProgramState {
-        const newState = BuilderState.clone(state) as BuilderProgramState;
-        newState.semanticDiagnosticsPerFile = state.semanticDiagnosticsPerFile && new Map(state.semanticDiagnosticsPerFile);
-        newState.changedFilesSet = new Set(state.changedFilesSet);
-        newState.affectedFiles = state.affectedFiles;
-        newState.affectedFilesIndex = state.affectedFilesIndex;
-        newState.currentChangedFilePath = state.currentChangedFilePath;
-        newState.currentAffectedFilesSignatures = state.currentAffectedFilesSignatures && new Map(state.currentAffectedFilesSignatures);
-        newState.currentAffectedFilesExportedModulesMap = state.currentAffectedFilesExportedModulesMap?.clone();
-        newState.seenAffectedFiles = state.seenAffectedFiles && new Set(state.seenAffectedFiles);
-        newState.cleanedDiagnosticsOfLibFiles = state.cleanedDiagnosticsOfLibFiles;
-        newState.semanticDiagnosticsFromOldState = state.semanticDiagnosticsFromOldState && new Set(state.semanticDiagnosticsFromOldState);
-        newState.program = state.program;
-        newState.compilerOptions = state.compilerOptions;
-        newState.affectedFilesPendingEmit = state.affectedFilesPendingEmit && state.affectedFilesPendingEmit.slice();
-        newState.affectedFilesPendingEmitKind = state.affectedFilesPendingEmitKind && new Map(state.affectedFilesPendingEmitKind);
-        newState.affectedFilesPendingEmitIndex = state.affectedFilesPendingEmitIndex;
-        newState.seenEmittedFiles = state.seenEmittedFiles && new Map(state.seenEmittedFiles);
-        newState.programEmitComplete = state.programEmitComplete;
-        return newState;
+    function backupEmitBuilderProgramState(state: Readonly<BuilderProgramState>): BackupEmitState {

```

Comment:
If it's the emit state of the `BuilderProgramState`, I would call it `backupBuilderProgramEmitState`.